scala/scala-impl/src/org/jetbrains/plugins/scala/project/Versions.scala

```diff
@@ -78,7 +80,7 @@ object Versions  {
     }
   }
 
-  private case class Entity(url: String, pattern: Regex, minVersion: Version, hardcodedVersions: Seq[String]) {
+  private case class Entity(var url: String, pattern: Regex, minVersion: Version, hardcodedVersions: Seq[String]) {
```

Comment:
This is painfully true if the object is placed in a `HashMap` or `HashSet`. The `hashCode` is calculated once based on the fields, but if they mutate, the hashcode would be different next time it's called. If the `Entity` was placed in a hash bucked based on the old version of the hashCode, it would never be found:

```
scala> case class Magic(var name: String)
defined class Magic

scala> val m = Magic("Jordan")
m: Magic = Magic(Jordan)
scala> val s = collection.mutable.HashSet(m)
s: scala.collection.mutable.HashSet[Magic] = Set(Magic(Jordan))

scala> s
res0: scala.collection.mutable.HashSet[Magic] = Set(Magic(Jordan))

scala> s.contains(m)
res1: Boolean = true

scala> m.name = "Air"
m.name: String = Air

scala> s
res2: scala.collection.mutable.HashSet[Magic] = Set(Magic(Air))

scala> s.contains(m)
res3: Boolean = false

scala> m
res4: Magic = Magic(Air)

scala> s.head == m
res7: Boolean = true
```

Notice that even though the set has one element, and that element is equal to `m`, `contains` is wrong.